File: Games/Uzta/src/main/kotlin/jokatu/game/games/uzta/views/UztaViewResolverFactory.java
package jokatu.game.games.uzta.views;

import jokatu.game.games.uzta.game.Uzta;
import jokatu.game.games.uzta.player.UztaPlayer;
import jokatu.game.viewresolver.ViewResolver;
import jokatu.game.viewresolver.ViewResolverFactory;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

/**
 * @author steven
 */
@Component
public class UztaViewResolverFactory extends ViewResolverFactory<UztaPlayer, Uzta> {
	@NotNull
	@Override
	protected Class<Uzta> getGameClass() {
		return Uzta.class;
	}

	@NotNull
	@Override
	protected ViewResolver<UztaPlayer, Uzta> getResolverFor(@NotNull Uzta castGame) {
		return new UztaViewResolver(castGame);
	}
}

File: Games/Uzta/src/main/kotlin/jokatu/game/games/uzta/views/UztaViewResolverFactory.kt
package jokatu.game.games.uzta.views

import jokatu.game.games.uzta.game.Uzta
import jokatu.game.games.uzta.player.UztaPlayer
import jokatu.game.viewresolver.ViewResolver
import jokatu.game.viewresolver.ViewResolverFactory
import org.springframework.stereotype.Component

/**
 * @author steven
 */
@Component
class UztaViewResolverFactory : ViewResolverFactory<UztaPlayer, Uzta>() {
	override val gameClass: Class<Uzta>
		get() = Uzta::class.java

	override fun getResolverFor(castGame: Uzta): ViewResolver<UztaPlayer, Uzta> {
		return UztaViewResolver(castGame)
	}
}
